src/Storages/Statistics/UniqStatistics.cpp

```diff
@@ -0,0 +1,63 @@
+#include <Storages/Statistics/UniqStatistics.h>
+#include <DataTypes/DataTypeNullable.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_STATISTICS;
+}
+
+UniqStatistics::UniqStatistics(const SingleStatisticsDescription & stat_, const DataTypePtr & data_type)
+    : IStatistics(stat_)
+{
+    arena = std::make_unique<Arena>();
+    AggregateFunctionProperties property;
+    property.returns_default_when_only_null = true;
+    uniq_collector = AggregateFunctionFactory::instance().get("uniq", NullsAction::IGNORE_NULLS, {data_type}, Array(), property);
+    data = arena->alignedAlloc(uniq_collector->sizeOfData(), uniq_collector->alignOfData());
+    uniq_collector->create(data);
+}
+
+UniqStatistics::~UniqStatistics()
```

Comment:
I don't know which one is better. Because RDMS databases usually use `histogram` `n_distinct`, see https://www.postgresql.org/docs/current/view-pg-stats.html

Maybe we should follow this style